symplyphysics/laws/thermodynamics/free_energy_differential.py

```diff
@@ -46,7 +51,57 @@
     -1 * entropy * temperature_change - pressure * volume_change + chemical_potential * particle_count_change,
 )
 
-# TODO: derive from the definition of Helmholtz free energy and internal energy differential
+# Derive from the definition of Helmholtz free energy and internal energy differential
+
+_internal_energy_sym = symbols("internal_energy", cls=SymFunction)
+_entropy_change = SymSymbol("entropy_change")
+_temperature = SymSymbol("temperature")
+_volume = SymSymbol("volume")
+_particle_count = SymSymbol("particle_count")
+
+_internal_energy = _internal_energy_sym(entropy, _volume, _particle_count)
+
+_free_energy = free_energy_def.law.rhs.subs({
+    free_energy_def.internal_energy: _internal_energy,
+    free_energy_def.temperature: _temperature,
+    free_energy_def.entropy: entropy,
+})
+
+_free_energy_change = (
+    _free_energy.diff(_temperature) * temperature_change
+    + _free_energy.diff(entropy) * _entropy_change
+    + _free_energy.diff(_volume) * volume_change
+    + _free_energy.diff(_particle_count) * particle_count_change
```

Comment:
What is this formula?

Reply:
Here I'm looking to find the differential of the free energy. I don't think that `sympy` can do it, so I have to do it by hand

Reply:
Add some hint about it.